snooty/main.py

```diff
@@ -322,8 +402,14 @@ def main() -> None:
             observer.join()
     except KeyboardInterrupt:
         pass
+    except:
+        if output_path:
+            os.unlink(output_path)
+        raise
     finally:
-        if connection:
+        if isinstance(backend, ZipBackend):
+            backend.close()
+        elif connection:
             print("Closing connection...")
             connection.close()
```

Comment:
It may be worth adding `.close` on the `MongoBackend` so we can have uniform responsibility of `close` across `Backend` derived classes, and then we can just call `backend.close` without needing to differentiate

something like 
```
def close(self) -> None:
     if(self.client):
         print("Closing connection...")
         self.client.close()
```
?

Reply:
Excellent idea, thank you!